File: backend_SB/src/main/java/tn/esprit/crmassurance/entities/Activity.java
package tn.esprit.crmassurance.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.Serializable;
import java.util.Date;

@Entity
@NoArgsConstructor
@Getter
@Setter
public class Activity implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Date ExchangeDate;
    private String information;

    @Enumerated(EnumType.STRING)
    private EActionAct action;

    @ManyToOne
    @JoinColumn(name = "lead_id", nullable = false)
    private User lead;

    @ManyToOne
    @JoinColumn(name = "representant_id", nullable = false)
    private User repAct;
}

File: frontend_Angular/src/app/model/Activity.ts
import { EActivity } from "./EActivity";
import { User } from "./User";

export class Activity{
id:number;
exchangeDate:Date;
information:string;
action:EActivity;
lead:any;
repAct:any;

constructor(act,inf,d){
   this.exchangeDate=d;
    this.information=inf;
    this.action=act;
}

setLead(l){
   this.lead=l
}

setRep(r){
    this.repAct=r
}

}